quickwit/quickwit-search/src/collector.rs

```diff
@@ -695,13 +699,149 @@ pub(crate) fn make_merge_collector(
     })
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+struct PartialHitSortingKey {
+    score: Option<SortValue>,
+    address: GlobalDocAddress,
+    asc: bool,
+}
+
+impl Ord for PartialHitSortingKey {
+    fn cmp(&self, other: &PartialHitSortingKey) -> Ordering {
+        assert_eq!(
+            self.asc, other.asc,
+            "comparing two PartialHitSortingKey of different ordering"
+        );
+        if self.asc {
+            self.score
+                .map(Reverse)
+                .cmp(&other.score.map(Reverse))
+                .then(other.address.cmp(&self.address))
```

Comment:
```suggestion
            self.score
                .map(Reverse)
                .cmp(&other.score.map(Reverse))
                .then(other.address.cmp(&self.address))
```

alternatively 

`(other.score, other.address).cmp(&(self.score, self.address))`

Reply:
(nitpick)